File: API/FoodCareApi/Models/Usuario.cs
namespace FoodCareApi.Models;


    public class Usuario{
        public int idUsuario{get;}
        public string nome  { get; set; } = string.Empty;
        public string email { get; set; } = string.Empty;
        public string senha { get; set; } = string.Empty;
        public string tipoPessoa {get; set;} = string.Empty;
        public string documento {get; set;} = string.Empty;
        public string? cep  { get; set; }
        public string? cidade { get; set; }
        public string? bairro { get; set; }
        public string? rua    { get; set; }
        public string? numero { get; set; }

        // Campos que a API externa vai preencher
        public double? latitude  { get; set; } 
        public double? longitude { get; set; }
          

    }



    

File: FoodCare/app/src/main/java/com/example/foodcare/model/Usuario.kt
package com.example.foodcare.model

data class CadastroRequest(
    val nome : String,
    val email: String,
    val senha: String,
    val tipoPessoa : String,
    val documento : String
)

data class CadastroResposta(
    val idUsuario: Int
)

data class DoadorRequest(
    val idUsuario: Int
)

data class ReceptorRequest(
    val idUsuario: Int
)

data class CompletarPerfilRequest(
    val nome: String,
    val email: String,
    val cep: String?,
    val cidade: String?,
    val bairro: String?,
    val rua: String?,
    val numero: String?,

    val latitude: Double?,
    val longitude: Double?
)
data class LoginRequest(
    val email: String,
    val senha: String
)

data class LoginResposta(
    val idUsuario: Int,
    val nome: String,
    val token: String,
    val tipoUsuario : String
)

data class Usuario(
    val nome: String,
    val email: String,
    val documento: String,
    val tipoPessoa: String,
    val cep: String?,
    val cidade: String?,
    val bairro: String?,
    val rua: String?,
    val numero: String?
)
